components/server/src/user/user-controller.ts

```diff
@@ -91,6 +92,39 @@ export class UserController {
             this.ensureSafeReturnToParam(req);
             await this.authenticator.authenticate(req, res, next);
         });
+
+        router.get("/login/ots/:user/:key", async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+            try {
+                const secret = await this.otsDb.get(req.params.key);
+                if (!secret) {
+                    res.sendStatus(401);
+                    return;
+                }
+
+                const user = await this.userDb.findUserByName(req.params.user);
```

Comment:
I'd be really in favor of a "lookup by Gitpod userId" because it's more precise, and thus _secure_. This be ok - if we manage it well! - for staging and prod, but never for all self-hosted installations.

Reply:
done